pkg/deploy/registry/rest.go

```diff
@@ -17,6 +17,41 @@ var (
 	ErrUnknownDeploymentPhase = errors.New("unknown deployment phase")
 )
 
+// WaitForRunningDeployer waits until the specified deployer pod name is
+// running.
+func WaitForRunningDeployer(pn kclient.PodsNamespacer, namespace, name string, timeout time.Duration) (bool, error) {
+	p, err := pn.Pods(namespace).Get(name)
+	if err != nil {
+		return false, err
+	}
+	if p.Status.Phase == kapi.PodRunning {
+		return true, nil
+	}
+	fieldSelector := fields.Set{"metadata.name": name}.AsSelector()
+	options := kapi.ListOptions{FieldSelector: fieldSelector, ResourceVersion: p.ResourceVersion}
+	w, err := pn.Pods(namespace).Watch(options)
```

Comment:
We now have watch.Until - you should use that instead of following what's going on in `WaitForRunningDeployment` (that should change to watch.Until too)